### PR TITLE
[AUT-3741] Corrected Ingresses configuration for mTLS

### PR DESCRIPTION
**Pull request by lisocki**
## Description

mTLS Ingress for ACP has relaxed requirements for the client certificate now. The new nginx conf:
```
ssl_verify_client                       optional_no_ca;
```

## Type of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Tests (extending the test suite)
- [ ] Refactor (internal improvement that doesn't change product functionality)
- [ ] Other (if none of the other choices apply)

### CODE REVIEW

**Review by mbilski — CHANGES_REQUESTED** [review]
If the mtls ingress is enabled this should be configured in acp as well

```
server:
  mtls_url: ""
```

This is also a breaking change for people who relies on optional_no_ca and depth 2 in default ingress.

We need a migration guide.

File: charts/acp/templates/configmap.yaml [hunk 1]
```diff
@@ -9,6 +9,9 @@ data:
   config.yaml: |
     server:
       url: {{ .Values.serverURL }}
+      {{- if .Values.ingressMtls.enabled }}
+      mtls_url: "{{ .Values.serverULRMtls }}"
```
Comment (mbilski) [inline charts/acp/templates/configmap.yaml:13]: ULR => URL

File: charts/acp/values.yaml [hunk 2]
```diff
@@ -22,6 +22,10 @@ imagePullSecrets:
 ##
 serverURL: "https://acp.local:8443"
 
+## Public ACP URL for Ingress working in mTLS
+##
+serverULRMtls: "https://mtls.acp.local:8443"
```
Comment (mbilski) [inline charts/acp/values.yaml:27]: ULR => URL

File: charts/acp/templates/ingress.yaml [hunk 3]
```diff
@@ -70,7 +67,7 @@ metadata:
     {{- include "acp.labels" . | nindent 4 }}
   annotations:
     kubernetes.io/ingress.class: nginx
-    nginx.ingress.kubernetes.io/auth-tls-verify-client: "on"
+    nginx.ingress.kubernetes.io/auth-tls-verify-client: "optional_no_ca"
```
Comment (mbilski) [inline charts/acp/templates/ingress.yaml:70]: let's do this instead

```
 nginx.ingress.kubernetes.io/server-snippet: |
      ssl_verify_client optional_no_ca;
      ssl_verify_depth 2;
```

the depth is required in some cases

**Review by mbilski — CHANGES_REQUESTED** [review]

File: charts/acp/templates/configmap.yaml (hunk 1 above)
Comment (lisocki) [reply charts/acp/templates/configmap.yaml]: Fixed

File: charts/acp/values.yaml (hunk 2 above)
Comment (lisocki) [reply charts/acp/values.yaml]: Fixed

File: charts/acp/templates/ingress.yaml (hunk 3 above)
Comment (lisocki) [reply charts/acp/templates/ingress.yaml:70]: I added the depth annotation.

**Review by mbilski — APPROVED** [review]

**Review by artursmolarek — APPROVED** [review]

**Pull request merged**